### PR TITLE
Gabbytang/find other profiles

### PR DESCRIPTION
**Pull request by gabbytang**
## Trello ticket link

[As a user, I want to be able to see all other profiles.](https://trello.com/c/SP78oY4X )
[As a user, I want to be able to open other users' profile pages.](https://trello.com/c/Ghs7YMqQ )
[As a user, I want to search for other user profiles of interest.](https://trello.com/c/r8FPPHpf )

## Implementation description
* Created Network page where users can see list of all active users
* Added search functionality to Network page so that users can search based on name 
* Each listed user has an embedded link which takes you to their profile page in a new window
* The profile pages just display all entered profile information currently in the db 
* If the selected user does not have profile information currently saved, clicking the link should still open a page, but there is a pop-up stating that the user has not set up their profile 
* The pop-up includes a button that takes the user back to their previous page by closing the new window 
* Added the Network page to the Navigation bar
*  EDITED - getUserProfile API but should not affect any other components that use it. Edited query to join userProfiles with user table to get names as well. 

## Steps to test
1. Ensure you can access the Network page via the nav bar 
2. Upon opening the network page for the first time, users should see the complete list of active users stored. 
4. Users should be able to type in the search bar at the top of the page 
5. Ensure that upon clicking "SEARCH", the list below should re-render to match the search term 
6. Check that only Active users are returned.
7. Users should also be able to click on the names of the listed users. 
8. Upon clicking the link, a new window should open
9. Check the new window is the profile page of the selected user
10. The listed information should be equal to that in the db
11.  Check that clicking a user with no profile info currently in the db opens up a new window with a dialog 
12. The dialog should include a button that returns the user to their previous page by closing the new window. 

## What should reviewers focus on?
* Make sure the search function is working as expected. Currently based off wildcards so you should be able to search for any subset of letters within a name and it should return the expected name (ex. "Ta"  should return "Gabrielle Tang")
* The links should lead to the appropriate profile pages 
* If a user does not have profile information, there should be a dialog that opens within the new popup window.
* Ensure the Profile page works as previously (that you can see/edit your own profile info  as I made a small change to one API it uses)

### CODE REVIEW

**Comment by c45choi** [comment]
gabby this is sooo good and cool.

**Review by c45choi — CHANGES_REQUESTED** [review]
* I can access the Network page via the nav bar
* Can see all active users
* Can type into the search bar
* Can see that upon clicking "SEARCH", the list below should re-render to match the search term
* Only Active users are returned.
* Can click on the names of the listed users.
* When I click the link, a new window should open
* The new window is the profile page of the selected user
* Clicking a user with no profile info currently in the db opens up a new window with a dialog
* The dialog includes a button that returns the user to their previous page by closing the new window.

Only question:
* Ensure the Profile page works as previously (that you can see/edit your own profile info as I made a small change to one API it uses) > not sure if this was like it before or if this is what you edited? I go into my profile from top right and i cant see my information in the boxes...

**Review by c45choi — CHANGES_REQUESTED** [review]
I also tried editing my profile and then checking my profile again and the changes were not updated

**Review by c45choi — APPROVED** [review]
works awesome bero thanks for the changes!

File: src/components/userNetwork/Network.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+const Network = () => {
+
+  const [profiles, setProfiles] = React.useState([]);
+  const [userSearchTerm, setUserSearchTerm] = React.useState('');
+  const handleUserSearch = (event) => {
+    setUserSearchTerm(event.target.value);
+  }
+
+  const loadGetUsers = async () => {
```
Comment (safchow) [inline src/components/userNetwork/Network.js]: this is not being used, I would remove it

File: src/components/userNetwork/NetworkProfile.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        display: "flex",
+    }
+}));
+
+const NetworkProfile = () => {
+
+    const classes = useStyles();
+    const { currentUser } = useAuth()
+    const [email, setEmail] = React.useState('');
+    const [firstName, setFirstName] = React.useState('');
```
Comment (safchow) [inline src/components/userNetwork/NetworkProfile.js]: some of these react states are never used, I would remove these

**Review by safchow — APPROVED** [review]
lgtm 👍 approved with comments

**Pull request merged**